.github/workflows/build.yml

```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
```

Comment:
Requirement was to have these checks run on push to main, but it may also be useful to have these things caught in the review pipeline (Reviewers should be running the tests themselves as well of course).

Reply:
Will add

```
pull_request:
    branches:
      - main
```